toolchain/sem_ir/typed_insts.h

```diff
@@ -717,7 +717,7 @@ struct InterfaceWitness {
 // Accesses an element of an interface witness by index.
 struct InterfaceWitnessAccess {
   static constexpr auto Kind =
-      InstKind::InterfaceWitnessAccess.Define<Parse::InvalidNodeId>(
+      InstKind::InterfaceWitnessAccess.Define<Parse::NodeId>(
```

Comment:
The `NodeId` is the parse node corresponding to the instruction. `InvalidNodeId` means there isn't one, plain `NodeId` means it could be a parse node of any kind. In this PR we start passing a location in when constructing an `InterfaceWitnessAction`, so there is now an associated parse node, so I'm using `NodeId` to mean "any parse node" rather than `InvalidNodeId` to mean "no parse node".

Reply:
Let's discuss when you're back. As I note on #4211, I don't agree with this interpretation, I don't think it reflects how the IR is working.

Reply:
Maybe I'm misunderstanding the existence of NodeId on some of these, actually, due to what's happening with constants and NoLoc. I'd not digested all of that, sorry. But still, we do have other instructions which use InvalidNodeId and get locations, particularly imports. Anyways, maybe #4211 is a better place to discuss.